src/graph/meter_roles.rs

```diff
@@ -0,0 +1,434 @@
+// License: MIT
+// Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+//! Methods for checking the roles of meters in a [`ComponentGraph`].
+
+use crate::{component_category::CategoryPredicates, ComponentGraph, Edge, Error, Node};
+
+impl<N, E> ComponentGraph<N, E>
+where
+    N: Node,
+    E: Edge,
+{
+    /// Returns true if a node is a grid meter.
+    ///
+    /// A meter is identified as a grid meter if:
+    ///   - it is a successor of the grid component,
+    ///   - all its siblings are meters,
+    ///   - if there are siblings, the successors of it and the successors of
+    ///     its siblings are meters.
```

Comment:
According to this specification, there can be multiple "grid meters", and the grid connection's total load would be the sum of all these meters.

This terminology seems a bit odd because if we call a meter a "grid meter", then we should only have to read this one meter to get the net grid activity.

Can we improve it?

Reply:
Wait, I thought we had cases like this, with multiple grid meters.  And you mentioned the "diamond" configuration for grid meters as well, right?  How do you suggest we change it?

But anyway, this particular method is there just so the existing SDK formulas can continue to work. It will go away as soon as the new approach for building formulas is implemented.  So we don't have to put in too much energy into this one.